camel-idea-plugin/src/main/java/org/apache/camel/idea/annotator/CamelAnnotator.java

```diff
@@ -0,0 +1,220 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.idea.annotator;
+
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import com.intellij.lang.annotation.AnnotationHolder;
+import com.intellij.lang.annotation.Annotator;
+import com.intellij.openapi.components.ServiceManager;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.psi.PsiElement;
+import org.apache.camel.catalog.CamelCatalog;
+import org.apache.camel.catalog.EndpointValidationResult;
+import org.apache.camel.catalog.SimpleValidationResult;
+import org.apache.camel.idea.catalog.CamelCatalogService;
+import org.apache.camel.idea.util.CamelService;
+import org.apache.camel.idea.util.IdeaUtils;
+import org.jetbrains.annotations.NotNull;
+
+import static org.apache.camel.idea.util.IdeaUtils.isEmpty;
+
+
+/**
+ * Validate Camel URI endpoint and simple expression and annotated the specific property
+ * or simple expression to highlight the error on the editor
+ */
+public class CamelAnnotator implements Annotator {
+    @Override
+    public void annotate(@NotNull PsiElement element, @NotNull AnnotationHolder holder) {
+        if (ServiceManager.getService(element.getProject(), CamelService.class).isCamelPresent()) {
+            String fromElement = IdeaUtils.extractTextFromElement(element);
+            if (fromElement != null && !fromElement.isEmpty() && !fromElement.trim().endsWith("&") && !fromElement.trim().endsWith("?")
```

Comment:
Yeah you may need to do a bit like this does: https://github.com/davsclaus/camel-idea-plugin/blob/master/camel-idea-plugin/src/main/java/org/apache/camel/idea/documentation/CamelDocumentationProvider.java#L272 however I am not 100% sure if the removal of & or ? is needed, but it may be. Certainly for the XML &amp; you need to unescape it.